Composer/packages/webchat-client/src/WebChatPanel.tsx

```diff
@@ -0,0 +1,178 @@
+
+      const webchatStore = createWebChatStore({});
+      const styleSet = createStyleSet({ ...webChatStyleOptions });
+      styleSet.fileContent = {
+        ...styleSet.fileContent,
+        background: `${NeutralColors.white}`,
+        '& .webchat__fileContent__fileName': {
+          color: `${CommunicationColors.primary}`,
+        },
+        '& .webchat__fileContent__size': {
+          color: `${NeutralColors.white}`,
+        },
+        '& .webchat__fileContent__downloadIcon': {
+          fill: `${NeutralColors.white}`,
+        },
+        '& .webchat__fileContent__badge': {
+          padding: '4px',
+        },
+      };
+
+      return (
+        <ReactWebChat
+          key={chatData.conversationId}
+          directLine={chatData.directline}
+          disabled={!botUrl}
+          store={webchatStore}
+          styleSet={styleSet}
+          userID={chatData.user.id}
+          activityMiddleware={conversationService.createActivityMiddleware}
+          cardActionMiddleware={conversationService.createCardActionMiddleware}
+          locale={activeLocale}
+        />
+      );
+    }
+    return null;
+  }, [chats]);
+
+  const onRestartConversationClick = async (oldConversationId: string, requireNewUserId: boolean) => {
+    const chatData = await conversationService.restartConversation(
+      chats[oldConversationId],
+      requireNewUserId,
+      activeLocale
+    );
+
+    setCurrentConversation(chatData.conversationId);
+
+    setChatData({
+      [chatData.conversationId]: chatData,
+    });
+  };
+
+  const onSaveTranscriptClick = async (conversationId: string) => {
+    try {
+      const downloadLink = downloadLinkRef.current;
+      if (!downloadLink) return;
+
+      const resp = await conversationService.getTranscriptsData(conversationId);
+      const transcripts = resp.data;
+
+      const blob = new Blob([JSON.stringify(transcripts, null, 2)], { type: 'application/json' });
+      const url = URL.createObjectURL(blob);
+      downloadLink.download = `${botName}.transcript`;
+      downloadLink.href = url;
+      downloadLink.click();
+    } catch (ex) {
+      //TODO: Handle failed transcript download
+    }
+  };
+
+  if (!chats[currentConversation]?.directline) {
+    return null;
+  }
+
+  return (
+    <>
+      <WebChatHeader
+        conversationId={currentConversation}
+        onRestartConversation={onRestartConversationClick}
+        onSaveTranscript={onSaveTranscriptClick}
+        openBotInEmulator={() => {
+          openBotInEmulator(projectId);
+        }}
+      />
+      <div data-testid="WebChat-Content" style={{ height: 'calc(100% - 36px)' }}>
+        {webchatContent}
+      </div>
+      {/* A shadow download link to trigger browser native API on saving transcript JSON.  */}
+      <a ref={downloadLinkRef} aria-hidden="true" href="#save" style={{ display: 'none' }}>
+        {formatMessage('Save')}
+      </a>
```

Comment:
If this is hidden in the UI, we don't even need to attach it to the dom until a transcript is downloaded.

```js
const btn = document.createElement('a');
btn.style.display = 'none';
someContainerRef.appendChild(btn);
// ... do download stuff
btn.click();
someContainerRef.removeChild(btn);
```